mercantile/__init__.py

```diff
@@ -137,11 +137,22 @@ def xy(lng, lat, truncate=False):
     -------
     x, y : float
     """
+    A = 6378137.0
+    R = A * math.pi
     if truncate:
         lng, lat = truncate_lnglat(lng, lat)
-    x = 6378137.0 * math.radians(lng)
-    y = 6378137.0 * math.log(
-        math.tan((math.pi * 0.25) + (0.5 * math.radians(lat))))
+    x = A * math.radians(lng)
+    try:
+        y = A * math.log(
+            math.tan((math.pi * 0.25) + (0.5 * math.radians(lat))))
+    except ValueError:
+        y = float('-inf')
+
+    if y > R:
+        y = R
+    elif y < -R:
+        y = -R
+
```

Comment:
I'm not willing to truncate y at `R` like this.

I would be in favor of return `float(inf)` for y when `lat >= 90.0` and `float(-inf)` for y when `lat <= -90`. I think that's more user-friendly than raising an exception as we do know.